pkg/processor/broadcast_storage_handler.go

```diff
@@ -81,7 +83,10 @@ func (h *BroadcastStorageHandler) TransactionConfirmed(
 	}
 
 	if err := h.parser.ExpectedOperations(intent, observed, false, true); err != nil {
-		return fmt.Errorf("confirmed transaction did not match intent: %w", err)
+		errMsg := fmt.Errorf("confirmed transaction did not match intent: %w", err)
+		if !isValidStakingOperation(intent, intentMetadata) {
```

Comment:
what if `confirmed transaction did not match intent` but staking operation is valid, then we won't catch the error?

Reply:
This code is used for check:construction only, we never full tested staking txns in check:construction.
Staking operation looks like this:
```{
  "network_identifier": {
    "blockchain": "abc",
    "network": "testnet"
  },
  "operations": [
    {
      "operation_identifier": {
        "index": 0
      },
      "type": "stake",
      "status": "",
      "account": {
        "address": "dydx12c0xh69avh8xd0z76as9as4mcy0cfncukmmd7z",
        "metadata": {}
      },
      "amount": {
        "value": "5000000",
        "currency": {
          "symbol": "COSMOSDYDX",
          "decimals": 18
        }
      }
    }
  ],
  "suggested_fee_multiplier": 2,
  "metadata": {
    "gas_adjustment": 0.5,
    "validator_src_address": "dydxvaloper10lzv79d96l7jh07z76ry6cnn6ftnnl8fgketzu",
    "validator_dst_address": ""
  }
}
```
The amount in the preprocess API is a positive number, but our block API interprets staking as two operations; the first one is negative. Hence we get an intent mismatch.